qtfred/src/mission/dialogs/ShipEditor/ShipWeaponsDialogModel.cpp

```diff
@@ -0,0 +1,379 @@
+#include "ShipWeaponsDialogModel.h"
+namespace fso {
+namespace fred {
+Banks::Banks(const SCP_string& _name, int aiIndex, int _ship, int multiedit, ship_subsys* _subsys)
+	: m_isMultiEdit(multiedit) ,name(_name), subsys(_subsys), initalAI(aiIndex), ship(_ship)
+{
+	aiClass = aiIndex;
+}
+void Banks::add(Bank* bank)
+{
+	banks.push_back(bank);
+}
+Bank* Banks::getByBankId(const int id)
+{
+	for (auto bank : banks) {
+		if (id == bank->getWeaponId())
+			return bank;
+	}
+	return nullptr;
+}
+SCP_string Banks::getName() const
+{
+	return name;
+}
+int Banks::getShip() const
+{
+	return ship;
+}
+ship_subsys* Banks::getSubsys() const
+{
+	return subsys;
+}
+bool Banks::empty() const
+{
+	return banks.empty();
+}
+const SCP_vector<Bank*> Banks::getBanks() const
+{
+	return banks;
+}
+int Banks::getAiClass() const
+{
+	if (name == "Pilot") {
+		return Ships[ship].weapons.ai_class;
+	} else {
+		return subsys->weapons.ai_class;
+	}
+}
+void Banks::setAiClass(int newClass)
+{
+	if (m_isMultiEdit) {
+		object* ptr;
+		int inst;
+		ptr = GET_FIRST(&obj_used_list);
```

Comment:
This could be simplified.
object* ptr = GET_FIRST(&obj_used_list);

and `inst` looks like it can be moved entirely inside the while loop.